src/otx/cli/utils/installation.py

```diff
@@ -0,0 +1,530 @@
+    """Parse requirements and returns torch, mmcv and task requirements.
+
+    Args:
+        requirements (list[Requirement]): List of requirements.
+
+    Raises:
+        ValueError: If torch requirement is not found.
+
+    Examples:
+        >>> requirements = [
+        ...     Requirement.parse("torch==1.13.0"),
+        ...     Requirement.parse("mmcv-full==1.7.0"),
+        ...     Requirement.parse("mmcls==0.12.0"),
+        ...     Requirement.parse("onnx>=1.8.1"),
+        ... ]
+        >>> parse_requirements(requirements=requirements)
+        (Requirement.parse("torch==1.13.0"),
+        [Requirement.parse("mmcv-full==1.7.0"), Requirement.parse("mmcls==0.12.0")],
+        Requirement.parse("onnx>=1.8.1"))
+
+    Returns:
+        tuple[Requirement, list[Requirement], list[Requirement]]: Tuple of torch, mmcv and other requirements.
+    """
+    torch_requirement: str | None = None
+    mm_requirements: list[str] = []
+    other_requirements: list[str] = []
+
+    for requirement in requirements:
+        if requirement.unsafe_name == "torch":
+            torch_requirement = str(requirement)
+            if len(requirement.specs) > 1:
+                warn(
+                    "requirements.txt contains. Please remove other versions of torch from requirements.",
+                    stacklevel=2,
+                )
+
+        elif requirement.unsafe_name in MM_REQUIREMENTS:
+            mm_requirements.append(str(requirement))
+
+        # Rest of the requirements are task requirements.
+        # Other torch-related requirements such as `torchvision` are to be excluded.
+        # This is because torch-related requirements are already handled in torch_requirement.
+        else:
+            # if not requirement.unsafe_name.startswith("torch"):
+            other_requirements.append(str(requirement))
+
+    if not torch_requirement:
+        msg = "Could not find torch requirement. OTX depends on torch. Please add torch to your requirements."
+        raise ValueError(
+            msg,
+        )
+
+    # Get the unique list of the requirements.
+    mm_requirements = list(set(mm_requirements))
+    other_requirements = list(set(other_requirements))
+
+    return torch_requirement, mm_requirements, other_requirements
+
+
+def get_cuda_version() -> str | None:
+    """Get CUDA version installed on the system.
+
+    Examples:
+        >>> # Assume that CUDA version is 11.2
+        >>> get_cuda_version()
+        "11.2"
+
+        >>> # Assume that CUDA is not installed on the system
+        >>> get_cuda_version()
+        None
+
+    Returns:
+        str | None: CUDA version installed on the system.
+    """
+    # 1. Check CUDA_HOME Environment variable
+    cuda_home = os.environ.get("CUDA_HOME", "/usr/local/cuda")
+
+    if Path(cuda_home).exists():
+        # Check $CUDA_HOME/version.json file.
+        version_file = Path(cuda_home) / "version.json"
+        if version_file.is_file():
```

Comment:
```suggestion
        if not version_file.is_file():
             return None
```

Reply:
Thanks for the suggestion.  I have fixed this. - https://github.com/openvinotoolkit/training_extensions/commit/39c65454f5bd5aeff26fbb60925e0f1064b993bb